[Windows][build]two Build failure on Windows on latest main branch

hi,  team,
     when i try to build the torchao with the latest code, it will meet below two failures,
1.
 _× python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [9 lines of output]
      W0603 21:46:20.257000 15024 site-packages\torch\distributed\elastic\multiprocessing\redirects.py:29] NOTE: Redirects are currently not supported in Windows or MacOs.
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 35, in <module>
        File "C:\work\ao\setup.py", line 571, in <module>
          long_description=open("README.md").read(),
        File "C:\ProgramData\miniforge3\envs\torchao_2025.1\lib\encodings\cp1252.py", line 23, in decode
          return codecs.charmap_decode(input,self.errors,decoding_table)[0]
      UnicodeDecodeError: 'charmap' codec can't decode byte 0x8f in position 10945: character maps to <undefined>
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed_

i have triaged the issue, and found that it is the https://github.com/pytorch/ao/commit/adc78b72cf6f8b1bb47b21d8fa525048e24f3497 which introduced this regression, and the root cause is there is a ⚠️ in the Readme.MD and windows can not decode this. the solution is either remove this ⚠️ or modify the setup.py with long_description=open("README.md").read() to long_description=open("README.md", encoding='utf-8').read(). 

2. after i fix the above failure with remove the ⚠️ in Readme.MD, i meet another failure like below,
_LINK : error LNK2001: unresolved external symbol PyInit__C
        C:\work\ao\build\temp.win-amd64-cpython-310\Release\torchao\csrc\rocm\swizzle\_C.lib : fatal error LNK1120: 1 unresolved externals
        error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2022\\BuildTools\\VC\\Tools\\MSVC\\14.43.34808\\bin\\HostX64\\x64\\link.exe' failed with exit code 1120_
i also triage this regression, and found it is introduced by https://github.com/pytorch/ao/commit/63f2e51bf034786a84259fe39e2e743285f51dda

could you please help to fix them?

@msaroufim thanks for the fix, what's the plan for the second issue "C:\work\ao\build\temp.win-amd64-cpython-310\Release\torchao\csrc\rocm\swizzle_C.lib : fatal error LNK1120: 1 unresolved externals"

--- a/setup.py
+++ b/setup.py
@@ -491,6 +491,15 @@ def get_extensions():
             print("Currently only gfx942 is supported. Compiling only for gfx942.")
         extra_compile_args["nvcc"].append("--offload-arch=gfx942")
         sources += rocm_sources
+    else:
+        # Remove ROCm-based sources from the sources list.
+        extensions_rocm_dir = os.path.join(extensions_dir, "rocm")
+        rocm_sources = list(
+            glob.glob(
+                os.path.join(extensions_rocm_dir, "**/*.cpp"), recursive=True
+            )
+        )
+        sources = [s for s in sources if s not in rocm_sources]
 
     use_cutlass = False
     cutlass_90a_sources = None

--- a/setup.py
+++ b/setup.py
@@ -495,9 +495,7 @@ def get_extensions():
         # Remove ROCm-based sources from the sources list.
         extensions_rocm_dir = os.path.join(extensions_dir, "rocm")
         rocm_sources = list(
-            glob.glob(
-                os.path.join(extensions_rocm_dir, "**/*.cpp"), recursive=True
-            )
+            glob.glob(os.path.join(extensions_rocm_dir, "**/*.cpp"), recursive=True)
         )
         sources = [s for s in sources if s not in rocm_sources]
 

